### ISSUE TITLE
Analyzer: Check arity of function calls

### ISSUE THREAD

**Issue by jenshaase** [jenshaase/phel-lang#6]
The analyzer should check if each function call has enough arguments given.

**Comment by Chemaclass** [comment]
Does this still need to be done? or is not relevant any longer, @jenshaase?

**Comment by jenshaase** [comment]
It is not a required feature and probably not that easy as I thought. I will evaluate it in the next days and if it is to complicate I will close the ticket.

**Comment by Chemaclass** [comment]
Maybe would be good to write a bit more details what do you expect from this issue, some context, reason behind and acceptance criteria. With that I might be able to help, or split it in subtasks if necessary. No rush

**Comment by jenshaase** [comment]
I just checked if it is possible and I found to many examples where it is not possible. To check this in all cases you basically have to build a type checker :)

I would probably be easier to catch the `ArgumentCountError` from PHP and provide a good error message. Currently it says:
`ArgumentCountError: Too few arguments to function class@anonymous::__invoke(), 1 passed in /tmp/__phelcO3k5t on line 4 and exactly 2 expected`

Therefore, I close this ticket.
